### PR TITLE
ticket 3

### PR DESCRIPTION
**Pull request by Decyl0**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const request = require("supertest");
+const app = require("../app.js");
+
+describe("/api/categories", () => {
+  test('GET - 200: responds with "message": "all ok"', () => {
```
Comment (cjplatten) [inline __tests__/app.test.js:5]: At this point there has not been a ticket for /api so this is out of the scope of what is currently expected to be in the app. 
There will be a /api ticket later that has different requirements that will conflict with this.

Please remove this test and the related code and resubmit this PR.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const request = require("supertest");
+const app = require("../app.js");
+
```
Comment (cjplatten) [inline __tests__/app.test.js:3]: Your database is not being seeded before each test, this will cause you problems in later tickets.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const { getCategories, newMessage } = require('./controllers/controller');
+
+const app = express();
+app.use(express.json());
```
Comment (cjplatten) [inline app.js:5]: This is not necessary until we need to add a body to our request for post or patch, please remove.

File: controllers/controller.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+exports.newMessage = (req, res, next) => {
+  res.send({ message: "all ok" }).status(200);
+};
+
+exports.getCategories = (req, res) => {
+  categories().then((category) => {
+    res.status(200).send({ category });
+  });
+};
```
Comment (cjplatten) [inline controllers/controller.js:11]: What happens in the case of an error in the promise?

File: app.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+app.get('/api/categories', getCategories);
+
+
+
+//ERROR HANDLING
+
+app.all("/*", (req, res) => {
+    res.status(404).send({ msg: "Invalid URL" });
+  });
+
```
Comment (cjplatten) [inline app.js:18]: How would you handle a 500 error?

File: controllers/controller.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const { categories } = require("../models/model");
+
+exports.newMessage = (req, res, next) => {
```
Comment (cjplatten) [inline controllers/controller.js:3]: see /api comments in test file

File: app.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const { getCategories, newMessage } = require('./controllers/controller');
+
+const app = express();
+app.use(express.json());
+
+app.get("/api", newMessage);
```
Comment (cjplatten) [inline app.js:7]: see /api comments in test file

**Review by cjplatten — COMMENTED** [review]
Please see comments and submit another PR for changes.

**Review by callumu21 — APPROVED** [review]
Looks good to merge! You may want to add an afterAll function to end your database connection to stop your terminal hanging at the end of tests. It's covered in the NC Notes if that helps 👍

**Pull request merged**